site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Every object returning by the annotated service method is able to convert to an HTTP response message by
+response converters, except for HttpResponse_ and AggregatedHttpMessage_ which are already in a
+form of response message. You can also write your own response converter by implementing
+ResponseConverterFunction_ as follows. Also similar to the request converter, you can call
+``ResponseConverterFunction.fallthrough()`` when your response converter is not able to convert the result
+to the HttpResponse_.
+
+.. code-block:: java
+
+    class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx,
+                                            Object result) throws Exception {
+            if (result == MyObject.class) {
+                return HttpResponse.of(HttpStatus.OK,
+                                       MediaType.PLAIN_TEXT_UTF_8,
+                                       "Hello, %s!", result.processedName());
+            }
+
+            // To the next response converter.
+            return ResponseConverterFunction.fallthrough();
+        }
+    }
+
+You can annotate your service method and class as follows.
+
+.. code-block:: java
+
+    @ResponseConverter(MyResponseConverter.class)
+    class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public MyObject hello() {
+            // MyResponseConverter will be used to make a response.
+            ...
+        }
+
+        @Post("/hola")
+        @ResponseConverter(MySpanishResponseConverter.class)
+        public MyObject hola() {
+            // MySpanishResponseConverter and MyResponseConverter will be used to make a response.
+            ...
+        }
+    }
+
+Armeria supports :ref:`media_type_nego`. So you may want to get a negotiated media type in order to set
+a ``Content-Type`` header on your response. In this case, you can access it in your response converter
+as follows.
+
+.. code-block:: java
+
+    class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx,
+                                            Object result) throws Exception {
+
+            MediaType mediaType = ctx.negotiatedProduceType();
+            if (mediaType != null) {
+                // Do something based on the media type.
+                ...
+            }
+            ...
+        }
+    }
+
+.. _more-about-converters-exception-handlers:
+
+More about converters and exception handlers
+--------------------------------------------
+
+There are 3 ways to specify converters and exception handlers, except that `@RequestConverter`_ annotates
+a variable. The first one is annotating a service method and the second one is annotating a service class,
+as you read the above sections. The last one is using ServerBuilder_ as follows.
```

Comment:
How about just renaming this section to:

```rst
Using ServerBuilder_ to configure converters and exception handlers
-------------------------------------------------------------------
```

and replacing this paragraph with:

```rst
You can specify converters and exception handlers using ServerBuilder_, without using the annotations explained in the previous sections::
```